### ISSUE TITLE
zmq_msg_sets analogous to zmq_msg_gets

### ISSUE THREAD

**Issue by metadings** [zeromq/libzmq#1305]
We've got

    int zmq_msg_get (zmq_msg_t *message, int property);
    const char *zmq_msg_gets (zmq_msg_t *message, const char *property);

And the respective

    int zmq_msg_set (zmq_msg_t *message, int property, int value);

Don't we need also a 

    int zmq_msg_sets (zmq_msg_t *message, const char *property, void *value);

or the like...?

**Comment by hintjens** [comment]
Yes, this is missing to allow arbitrary metadata to be sent to a peer.

**Comment by rodgert** [comment]
I believe socket_base.cpp:798 currently precludes the sending of metadata.
This was introduced as a fix for issue #1058
<https://github.com/zeromq/libzmq/issues/1058>.
